jena-rdfconnection/src/main/java/org/apache/jena/rdfconnection/RDFConnectionRemote.java

```diff
@@ -492,8 +495,33 @@ protected HttpEntity graphToHttpEntity(Graph graph) {
         return graphToHttpEntity(graph, outputTriples);
     }
 
-    /** Create an HttpEntity for the graph */
+    /** Create an HttpEntity for the graph. */
     protected HttpEntity graphToHttpEntity(Graph graph, RDFFormat syntax) {
+        // Length - leaves connection reusable. 
+        return graphToHttpEntityWithLength(graph, syntax);
+    }
+    
+    /** 
+     * Create an HttpEntity for the graph. The HTTP entity will have the length but this
+     * requires serialising the graph at the point when this function is called.  
+     */
+    private HttpEntity graphToHttpEntityWithLength(Graph graph, RDFFormat syntax) {
+        String ct = syntax.getLang().getContentType().getContentType();
+        ByteArrayOutputStream out = new ByteArrayOutputStream(128*1024);
```

Comment:
No huge difference, but just out of curiosity, why not `try-with-resource` here?

Reply:
Needs to pass the ByteArrayOutputStream out of the "try" scope and until Java9 you can't write `try (out) {}` so it's a bit yukky.

And I added the "IO.close" for neatness as it is not really needed.